### PR TITLE
Search4e Notebook Merge Added Water Pouring Problem

### PR DESCRIPTION
**Pull request by himanshupathak21061998**
@MrDupin please check these Trying to solve issue#569

### CODE REVIEW

File: search.py [hunk 1]
```diff
@@ -984,6 +984,54 @@ def mutate(x, gene_pool, pmut):
...
+# _____________________________________________________________________________
+# Implementation of Water Pouring Problem
+class PourProblem(Problem):
+    """Problem about pouring water between jugs to achieve some water level.
+    Each state is a tuples of levels. In the initialization, provide a tuple of 
+    capacities, e.g. PourProblem(capacities=(8, 16, 32), initial=(2, 4, 3), 
+    goals={7}), which means three jugs of capacity 8, 16, 32, currently filled 
+    with 2, 4, 3 units of water, respectively, and the goal is to get a level 
+    of 7 in any one of the jugs."""
+    def __init__(self, initial = None, goals = (), capacities = None):
```
Comment (ad71) [inline search.py]: we usually skip spaces between `=` in the argument lists, as per pep8.

**Comment by himanshupathak21061998** [comment]
@ad71 I corrected those mistakes

**Comment by himanshupathak21061998** [comment]
Is there anything to improve in this PR because it is not yet merger @MrDupin
@ad71

**Comment by himanshupathak21061998** [comment]
Is there anything needed to improve in this PR because it is not yet merged@MrDupin
.@ad71

**Comment by himanshupathak21061998** [comment]
Is there any thing more is to be done on this PR ?
because it is not yet merged @MrDupin @ad71

**Comment by ad71** [comment]
@himanshupathak21061998 substantial pull requests can only be merged by @norvig . You'll have to wait until he finds the time to review your PR. MrDupin merges the minor pull requests and can not merge this without Dr. Norvig's consent.

**Pull request closed without merging**